### ISSUE TITLE
Bug Report: Flakiness in Local example

### ISSUE THREAD

**Issue by GuptaManan100** [vitessio/vitess#11295]
### Overview of the Issue

Ever since VTOrc has been introduced in the local example, the workflow that tests this has been very flaky.

Some of the failed runs - 
- https://github.com/vitessio/vitess/actions/runs/3093773569/jobs/5012933857
- https://github.com/vitessio/vitess/actions/runs/3076563628/jobs/4970806413

### Reproduction Steps

Check the actions and find the failed workflows for `local_example`.

### Binary Version

```sh
main
```

### Operating System and Environment details

```sh
main
```

### Log Fragments

```sh
E0921 10:14:29.095522   15738 main.go:47] rpc error: code = Unknown desc = tablet zone1-0000000402 InitReplica failed: rpc error: code = DeadlineExceeded desc = context deadline exceeded;tablet zone1-0000000401 InitReplica failed: rpc error: code = DeadlineExceeded desc = context deadline exceeded
```

**Comment by GuptaManan100** [comment]
I have spent considerable time investigating this flakiness and I have found 2 issues!

Numero Uno - 
VTOrc runs `ClusterHasNoPrimary` recovery just as it sees the first tablet of a new shard is up. This leads to issues since the durability_policy is semi-sync, so the `PRS` call doesn't succeed and actually blocks on `PopulateReparentJournal` and finally times out. 
Following this, we run `InitShardPrimary` from the local_example script, and that makes things even worse, since `ResetReplication` call on the previously promoted primary also fails. This is because it is stuck on semi-sync ACK and that also blocks `RESET MASTER` calls.

Numero Dos -
VTOrc runs `ClusterHasNoPrimary` and promotes a new primary successfully. The primary starts initializing the backend schema in vttablets, including `schema_migrations` tables which have a bunch of alter queries.
Before all of these queries succeed, we run `InitShardPrimary` which resets replication on all the tablets. This doesn't stop the writes on the primary. We then take the primary's replication position and paste them on the replica's without replicating anything. Then when we try to start replicaiton, it fails complaining 
```
2022-09-21T12:19:45.388093Z 46 [ERROR] [MY-010584] [Repl] Slave SQL for channel '': Worker 1 failed executing transaction 'a4f5dc7c-39a7-11ed-b071-3ed77ce818bb:7' at master log vt-0000000400-bin.000001, end_log_pos 3281; Error 'Table '_vt.schema_migrations' doesn't exist' on query. Default database: ''. Query: 'ALTER TABLE _vt.schema_migrations add column retries int unsigned NOT NULL DEFAULT 0', Error_code: MY-001146
2022-09-21T12:19:45.388247Z 45 [ERROR] [MY-010586] [Repl] Error running query, slave SQL thread aborted. Fix the problem, and restart the slave SQL thread with "SLAVE START". We stopped at log 'vt-0000000400-bin.000001' position 3037
```
This causes replication to be permanently broken and `ISP` to fail.
